src/Microsoft.AspNetCore.SignalR.Client.TS/Formatters.ts

```diff
@@ -0,0 +1,146 @@
+﻿import { Message, MessageType } from './Message';
+
+let knownTypes = {
+    "T": MessageType.Text,
+    "B": MessageType.Binary,
+    "C": MessageType.Close,
+    "E": MessageType.Error
+};
+
+function splitAt(input: string, searchString: string, position: number): [string, number] {
+    let index = input.indexOf(searchString, position);
+    if (index < 0) {
+        return [input.substr(position), input.length];
+    }
+    let left = input.substring(position, index);
+    return [left, index + searchString.length];
+}
+
+export namespace SSE {
```

Comment:
> In case of long polling you could keep state inside the parser itself instead of outside.

There isn't any parser state held outside the parser itself. The only exported function `parseMessages` expects a complete input and returns a complete output. **Internal** functions use tuples to avoid having to slice strings and create a bunch of new string objects, but otherwise, all the state is self-contained in the single public API.